lms/djangoapps/program_enrollments/rest_api/v1/serializers.py

```diff
@@ -104,7 +104,11 @@ def get_curriculum_uuid(self, obj):
         return str(obj.program_enrollment.curriculum_uuid)
 
     def get_course_staff(self, obj):
-        return is_course_staff_enrollment(obj)
+        # TODO: remove is_course_staff_enrollment check once course_roles is fully impelented and data is migrated
+        return (
+            is_course_staff_enrollment(obj) or
+            obj.program_enrollment.user.has_perm(CourseRolesPermission.MANAGE_STUDENTS.perm_name)
```

Comment:
Can you clarify a bit on how the new permission check. It appears to be different from the existing is_course_staff_enrollment which checks for a role on the course specifically. How does this new line know the user has MANAGE_STUDENTS for this course?

Reply:
In the new CourseRoles system a role is made up of permissions. The CourseRoles roles can be assigned directly on a course, but they can also be assigned on an organization or for the entire instance. In the case where it is assigned on the organization or the instance the role is granting the course level permissions access for all courses that belong to the org or instance.

This check will check if the user has the permission for the course, the org the course belongs to, or the instance the course belongs to. The permissions check is defined using the bridgekeeper library which allows for extending the `has_perm` function to make the specific db check necessary to confirm the user has the permissions for the course.